crates/compiler/can/src/operator.rs

```diff
@@ -609,20 +609,27 @@ fn desugar_bin_ops<'a>(
     let mut op_stack: Vec<Loc<BinOp>> = Vec::with_capacity_in(lefts.len(), arena);
 
     for (loc_expr, loc_op) in lefts {
-        arg_stack.push(desugar_expr(arena, loc_expr));
+        arg_stack.push(desugar_expr_keep_parens(arena, loc_expr));
         match run_binop_step(arena, whole_region, &mut arg_stack, &mut op_stack, *loc_op) {
             Err(problem) => return problem,
             Ok(()) => continue,
         }
     }
 
-    let mut expr = desugar_expr(arena, right);
+    let mut expr = desugar_expr_keep_parens(arena, right);
 
     for (left, loc_op) in arg_stack.into_iter().zip(op_stack.into_iter()).rev() {
         expr = arena.alloc(new_op_call_expr(arena, left, loc_op, expr));
     }
 
-    expr
+    desugar_expr(arena, expr)
+}
+
+fn desugar_expr_keep_parens<'a>(arena: &'a Bump, expr: &'a Loc<Expr<'a>>) -> &'a Loc<Expr<'a>> {
+    match expr.value {
+        ParensAround(_) => expr,
```

Comment:
Having this special case for parens feels unnatural. As I understand, the fix just changes the precedence level for parens, but it's probably not the correct place for that.
Various other attempts to solve the bug caused `roc_load` to hang, making debugging virtually impossible.

Reply:
Also, here I just ignored the inner expr. Does it make sense to desugar it at this stage?

Reply:
Updated in https://github.com/roc-lang/roc/pull/5481/commits/9f121d7bd1532fdd92275c08bd3964c9296f2353